### ISSUE TITLE
I Have a some questions

### ISSUE THREAD

**Issue by girlsending0** [NeuSpeech/NeuSpeech1#1]
hi
I am a researcher studying EEG-To-Text. I recently saw your Neuspeech paper. I was impressed by your paper, and it was a great help to my research direction. thanks. But I have some kinds of questions.

Have you ever applied ZuCO data? Did your model perform poorly on EEG data? 
I am also trying various things using ZuCo data, but I am having a lot of trouble because I am not getting any meaningful results.

Could I possibly get your baseline code? I would like to experiment with MEG data using your dataloader and code. I look forward to a positive response.
thank you

I am very happy to be doing the same research as you.😄😄

**Comment by NeuSpeech** [comment]
Hi, it is a pleasure to chat with you. I did not try my model on ZuCo dataset, but I tried eeg-to-text on ZuCo https://github.com/NeuSpeech/EEG-To-Text/tree/master. The results maybe consistent with yours. To avoid another misleading in the field, we are doing more experiments to ensure the correctness of our research. But we are not far from that, and all code will be released within one month.

**Comment by girlsending0** [comment]
@NeuSpeech Thank you for your kind reply.

Is it feasible to conduct experiments with the ZuCo dataset using your model? If not, could you please explain the reasons behind it?

Additionally, in the evaluation of the MEG dataset with the EEGToText model, it was mentioned that word-level features were employed for the MEG signal. I'm intrigued to know the specifics of these features. Were they derived from the average value of the MEG signal at the word level? I'm also curious about whether the raw signal itself was directly input into the model, or if there was a specific process to transform it into word-level features.

I appreciate that we're pursuing similar research endeavors. I eagerly await your response. Thank you very much.😄

**Comment by NeuSpeech** [comment]
@girlsending0 Thanks for your curiosity. 
1, It is feasible to conduct experiments with the ZuCo dataset using my model, it is just because me and my co-author does not think it is necessary to do it.
2, It is because the EEGTOTEXT model needs word-level segmentation to extract word features using traditional methods. It is actually the DE feature calculated from some bands. For more details, you can read ZuCo dataset paper and their dataset repo which has the code to extract these features at https://osf.io/q3zws/files/osfstorage and https://osf.io/2urht/files/osfstorage
3, In our model, we only input the raw signal without transforming it into word-level features.
I am happy that we are pursuing same goal, if you are interested, you can join us to do some research together. 🙌

**Comment by girlsending0** [comment]
@NeuSpeech 
Thank you for your detailed answer. 
I apologize for the inconvenience, but I have a few more questions and would appreciate your answers.

1. It is feasible to conduct experiments with the ZuCo dataset using my model, it is just because me and my co-author does not think it is necessary to do it.
=> Why do you think EEG data is not needed? Is it because you decided that ZuCo's EEG wasn't enough to decode the language?

2. It is because the EEGTOTEXT model needs word-level segmentation to extract word features using traditional methods. It is actually the DE feature calculated from some bands. For more details, you can read ZuCo dataset paper and their dataset repo which has the code to extract these features at https://osf.io/q3zws/files/osfstorage and https://osf.io/2urht/files/osfstorage
=> The dataset contains the EEG signal itself, not features. Therefore, you can use it in your Neuspeech model. I am conducting research using the EEG signal. Also, the features shown in ZuCo are known to be the average value of the amplitude information obtained after hilbert transform. Is it correct that DE stands for differential entropy?
![image](https://github.com/NeuSpeech/NeuSpeech1/assets/83329561/c207f1e3-d386-4b9b-8384-bf75532a4737)
![image](https://github.com/NeuSpeech/NeuSpeech1/assets/83329561/9efbcf09-6fc2-41e2-8435-b043acd018ac)


3. In our model, we only input the raw signal without transforming it into word-level features.
=> If word-level raw MEG signals were inserted, the time would obviously be different for each word. How did you adjust the shape?
And is the MEG signal of one sentence input to the whisper model at each timepoint, or is it segmented and input at each word?


Also, a question arose while reproducing your paper.

4. When using the GWilliams dataset, how did you divide it into train, test, and validation (8:1:1)?
It appears that the Ablation study was a separate experiment, so I am curious about what rules were set and divided in the main. Is the data divided randomly?

5. And, as described in the paper, the number of data in the GWilliams dataset is 661, but when I checked it myself, there were a total of 602 (ixi: 47, cable_spool_fort: 117, easy_money: 217, The_black_willow: 221).
Can you tell me under what conditions the data was extracted?

![image](https://github.com/NeuSpeech/NeuSpeech1/assets/83329561/4f6d76c7-d992-4df7-b34f-e531a44f48e3)



**If I can, I think it would be good to study together. Is there any way I can participate in the research?**
However, one thing to consider is that I am receiving a master's degree next year february and will be graduating soon.



Thank you very much for your suggestion and kind reply.😃😃

**Comment by dongyangli-del** [comment]
@girlsending0  @NeuSpeech After carefully reading the original paper, I almost definitely think that the authors exaggerated their results in the paper, especially the performance of w/o teacher forcing.

I edited my question into an email and sent it to the author team, and now it's been three weeks and still no response.

**Comment by girlsending0** [comment]
@dongyangli-del 

We are aware of that situation, and the original authors of the paper are also aware of it. It seems that referring to https://github.com/NeuSpeech/EEG-To-Text/tree/master would be appropriate.

**Comment by NeuSpeech** [comment]
Sorry for not replying. We didn't exaggerate the results, which are reproducible, but it does not mean it has no problem.

**Comment by Meirtz** [comment]
@girlsending0 
I've spent some time using the Whisper model to train on the ZuCo dataset, but the conclusions align closely with your recent findings posted on arXiv: it seems futile to use the ZuCo dataset for EGGToText tasks, as the results post-training are indistinguishable from noise. I also tried training models using the approach proposed in DeWave to convert EEG signals into spectrograms, but without success. Additionally, I've identified significant issues with the current methods of generating spectrograms, and segmenting EEG based on eye movement data does not appear effective. I read your article on arXiv yesterday—it was very intriguing! I am now working on further summarizing and identifying why previous methods did not work(currently, I believe there are significant problems with the prior feature extraction work).

**Comment by girlsending0** [comment]
@Meirtz 
Thank you for reading my paper with interest.
I think that the Word-level EEG features provided by ZuCo may not be suitable for text generation. Previously in my project, I tried various methods using the raw EEG signals from the ZuCo dataset, but the performance was not good.

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 

Hi! 
Thank you for the wonderful research and for sharing the code!
I would like to conduct a replication experiment, but I have some questions due to certain unclear points:

- Regarding the split, it seems there are two different methods. Which method did you ultimately adopt?
- Concerning the split, it is currently difficult to achieve complete replication as the seed is not fixed. Could you please share the actual split that was used?
- There are a lot of commands listed for fine-tuning. Could you please tell me which commands were specifically used in the paper?

If there is any additional information necessary for replication, please let me know.
Thank you in advance.

**Comment by NeuSpeech** [comment]
@YutoNishimura-v2 Thanks for your attention on our project, and thanks for raising issue. I am so sorry that I didn't clean this repo and makes it confusing. I cleaned code this morning, and set the seed. All cmd are in readme. Hope you can follow the readme smoothly this time. Feel free to raise any issues!

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 

Thank you very much for your prompt response!! It is very clear and easy to understand!
Is my understanding correct that this random seed is exactly the same as the one you used in your research?
It would be very helpful if you could provide the jsonl file that was actually used.
Thank you in advance.

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 

Hi, I could finally utilize your codes! Thx!
Based on your README, I trained & evaluated the model and the result is this:
`{"bleu-1": 0.7079590559005737, "bleu-2": 0.6943422555923462, "bleu-3": 0.6862793564796448, "bleu-4": 0.6807493567466736, "mer": 0.34607309103012085, "rouge1_fmeasure": 68.13008117675781, "rouge1_precision": 68.74004364013672, "rouge1_recall": 68.68816375732422, "rouge2_fmeasure": 62.207210540771484, "rouge2_precision": 62.777130126953125, "rouge2_recall": 62.82401657104492, "rougeL_fmeasure": 67.9247817993164, "rougeL_precision": 68.5459976196289, "rougeL_recall": 68.4614028930664, "rougeLsum_fmeasure": 67.9280014038086, "rougeLsum_precision": 68.54884338378906, "rougeLsum_recall": 68.46511840820312, "wer": 0.3681686818599701, "wil": 0.5148532390594482, "wip": 0.48514679074287415, "bert_p": 93.859375, "bert_r": 93.73838806152344, "bert_f": 93.78557586669922, "meteor": 0.6440672739306849}`

It seems that the result is really better than the result which your paper showed.
Do you know why? I guess the reason is the difference of split. 
Again, please share your split json with us.

Also, there is no code for evaluation of whisper with speech data. To complete reproducing, please share it, too.

Best,

**Comment by girlsending0** [comment]
@YutoNishimura-v2 How to set your split setting? I want to know your detailed split setting

**Comment by YutoNishimura-v2** [comment]
Thank you for your reply.
I just used gwilliams2023_process.py.
So the seed is fixed by 42.

They did not fix the seed at first, so I think the split is different from them.

I uploaded the test split I'm using.
[test.txt](https://github.com/user-attachments/files/16406554/test.txt)

**Comment by NeuSpeech** [comment]
@YutoNishimura-v2 thanks for your experiments. Actually, I am sorry that I didn't set seed in my previous experiments since I do not know about it. It is OK to share the test split, but it would be different from yours and might still be difficult to split accordingly, but I will provide in this comment.
[test.txt](https://github.com/user-attachments/files/16420916/test.txt)

As for speech evaluation, I used this repo https://github.com/yeyupiaoling/Whisper-Finetune to evaluate speech performance, so I do not know if it is appropriate to put the code in my repo. But you can use this repo to evaluate the results quickly.

Best,

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 

Great thank you for your support.
I'm so sorry to ask again, but could you share valid split, too? 

About speech evaluation, got it. I will try it later!

Best,

**Comment by NeuSpeech** [comment]
@YutoNishimura-v2 ok, here is the valid split
[val.txt](https://github.com/user-attachments/files/16422923/val.txt)

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 
Thank you very much!

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 
I checked the whisper's code. However, I think I should use the exactly same code with you.
If possible, could you share the evaluation code for speech?
Thank you.

**Comment by NeuSpeech** [comment]
@YutoNishimura-v2 ok, I will do that when I have spare time.

**Comment by YutoNishimura-v2** [comment]
@NeuSpeech 
Hi! How are you doing since then? I know you are busy, but your response would be greatly appreciated. Thank you very much.

**Comment by NeuSpeech** [comment]
> @NeuSpeech Hi! How are you doing since then? I know you are busy, but your response would be greatly appreciated. Thank you very much.

Thanks for your attention. I am so busy lately, so I didn't reply you. But I updated evaluation speech and how to use just now, you can check out readme.md in the last paragraph.

**Comment by YutoNishimura-v2** [comment]
Thank you very much!!

This is what I changed to use it:
- Delete "en_cer"
- Add "os.makedirs(outputdir,exist_ok=True)"

Anyway, I got a result! 
Again, thank you very much.
Best,

**Comment by NeuSpeech** [comment]
okokokok, thanks

**Comment by kunkunnai** [comment]
> [@NeuSpeech](https://github.com/NeuSpeech)
> 
> 非常感谢您的支持。很抱歉再次询问，但您也能分享有效的拆分吗？
> 
> 关于语音评估，明白了。我稍后再试！
> 
> 最好
Hello, can you please tell me why your test data is 1171 and the val data is 2917, which does not seem to be in line with 8:1:1. Thank you very much!
